### PR TITLE
Returning a reference from various corestr.cpp calls to avoid unnecessary string copies

### PR DESCRIPTION
**Pull request by npwoods**

### CODE REVIEW

File: src/lib/util/corestr.cpp [hunk 1]
```diff
@@ -222,7 +222,7 @@ std::string strmakeupper(std::string& str)
  * @return  A std::string.
```
Comment (cuavas) [inline src/lib/util/corestr.cpp]: Could you make this doxy comment a bit more useful?

The `@brief` description just "verbises" the function name - if you don't know what it does, this isn't going to help.

For the `@return` entry, firstly it returns a reference to a `std::string`, secondly the comment should not just repeat what one can see in the declaration, and thirdly it should hint at the significance of the returned string reference.

**Comment by cuavas** [comment]
Also, have you checked all uses of these functions?  Not just by running MAME a couple of times, actually checking every use of these functions in the codebase.  This will semantically change this kind of code:

```
    std::string str = "abc";
    std::string const &upper = strmakeupper(str);
    str = "def";
    // before this change, at this point (upper == "ABC") && (str == "def")
    // after this change, at this point (upper == "def") && (str == "def")
```

The reason being if you construct an automatic const reference variable with value return, it will bind to an anonymous variable with the same scope.  After the change, the function returns a reference, so `upper` will just be a const reference to `str`.  Checking for fallout from these kinds of subtle semantic changes is not always easy - it requires you to think about the pitfalls and read a lot of code.

**Comment by npwoods** [comment]
There is not a single use of these functions that isn't either:
1. Using the results "temporarily" in an expression
2. Ignoring the results outright

**Comment by cuavas** [comment]
OK, cool.

**Pull request merged**